AppSyncRealTimeClient/Interceptor/RealtimeGatewayURLInterceptor.swift

```diff
@@ -23,11 +23,18 @@ public class RealtimeGatewayURLInterceptor: ConnectionInterceptor {
         guard var urlComponents = URLComponents(url: request.url, resolvingAgainstBaseURL: false) else {
             return request
         }
+        
         urlComponents.scheme = SubscriptionConstants.realtimeWebsocketScheme
-        urlComponents.host = host.replacingOccurrences(
-            of: SubscriptionConstants.appsyncHostPart,
-            with: SubscriptionConstants.appsyncRealtimeHostPart
-        )
+        if(AppSyncURLHelper.hasStandardAppSyncGraphQLDomain(url: endpoint)) {
```

Comment:
what's the difference between the `request` and `endpoint`? Would we be able to tell if it is a standard AppSync domain endpoint from the `urlComponents`?

Reply:
`request` is an instance of `AppSyncConnectionRequest` which is a wrapper around `URL`. I think this wrapper must have been written for `ConnectionInterceptable` but currently there is not much difference between `request` and `endpoint`. 
`URLComponents` is convenient class to help modify host/path in a `URL` and then reconstruct the final `URL`. For checking a standard appsync domain endpoint, the regex needs to be applied on the complete urlstring.

Reply:
The reason i'm asking is because regex are complicated to understand and hard to maintain. If iOS platform has the URL easily acccessible down to it's components, aka `URLComponets`, is it easily achievable to swap the implementation over to use the components so the we can make a decision to remove the regex? Does URLComponents of the standard domain have the individual components (can we print the properties out?) such that you can do individual checks with the following:
1. It has a subdomain of 26 letters
2. Its domain matches `appsync-api`
3. It's 3rd domain component is in the format of a region.
4. the host is amazonaws
5. ".com" check?
6. path check on `/graphql` ?

Reply:
discussed this, and `AppSyncConnectionRequest` is gets reduced using multiple connection interceptors, `endpoint` is passed in everytime. for the second point, let's keep the regex since URLcompoents doesn't give sub host components, and keep platform's all aligned using the same thing (the regex checking)